src/tool/python/code_writers.h

```diff
@@ -1,60 +1,92 @@
 #pragma once
 
-namespace xlang
+namespace pywinrt
 {
-    void write_import_type(writer& w, TypeDef const& type)
+    template<typename F>
+    void write_snake_case(writer& w, std::string_view const& name, F case_func)
     {
-        if (is_exclusive_to(type))
+        XLANG_ASSERT(name.size() > 0);
+
+        for (std::string_view::size_type i = 0; i < name.size(); i++)
         {
-            return;
+            if (isupper(name[i]) && i > 0)
+            {
+                //TODO: special case IPAddress so that it doesn't get emitted as i_p_address
+                auto sub = name.substr(i - 1, 4);
+                if (sub != "UInt" && sub[0] != '_')
```

Comment:
Seems you're going to need a lot of special casing for edge cases.

Reply:
I don't think so, but I updated the code to support easily adding more exceptions